From 7b5277ab671e5b4a34d09ef60977cd7705358847 Mon Sep 17 00:00:00 2001
From: Double Lo <dblo@cypress.com>
Date: Thu, 20 Sep 2018 01:44:01 -0500
Subject: [PATCH 49/75] brcmfmac: configure wowl parameters in suspend function
 only if firmware support wowl

This patch removes the redundant wowl configuration for none wowl
FW.

Signed-off-by: Lo-Hsiang Lo <double.lo@cypress.com>
---
 drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
index 884a23320e77..5c9f4bacae6f 100644
--- a/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
+++ b/drivers/net/wireless/broadcom/brcm80211/brcmfmac/cfg80211.c
@@ -3761,8 +3761,9 @@ static s32 brcmf_cfg80211_suspend(struct wiphy *wiphy,
 		brcmf_set_mpc(ifp, 1);
 
 	} else {
-		/* Configure WOWL parameters */
-		brcmf_configure_wowl(cfg, ifp, wowl);
+		if (brcmf_feat_is_enabled(ifp, BRCMF_FEAT_WOWL))
+			/* Configure WOWL parameters */
+			brcmf_configure_wowl(cfg, ifp, wowl);
 	}
 
 exit:
-- 
2.20.1

